smalltalksrc/VMMaker/SpurMemoryManager.class.st

```diff
@@ -9712,11 +9783,22 @@ SpurMemoryManager >> positiveMachineIntegerValueOf: integerValue [
 	^coInterpreter positiveMachineIntegerValueOf: integerValue
 ]
 
+{ #category : #'store check' }
+SpurMemoryManager >> possiblePermStoreInto: destObj [
+
+	<inline: true>
+	((self isRemembered: destObj) and: [ 
+		 oldSpaceRememberedSet isInRememberedSet: destObj ]) ifFalse: [ 
+		oldSpaceRememberedSet remember: destObj ]
+]
+
 { #category : #'store check' }
 SpurMemoryManager >> possibleRootStoreInto: destObj [
+
 	<inline: true>
-	(self isRemembered: destObj) ifFalse:
-		[scavenger remember: destObj]
+	((self isRemembered: destObj) and: [ 
+		 newSpaceRememberedSet isInRememberedSet: destObj ]) ifFalse: [ 
+		newSpaceRememberedSet remember: destObj ]
```

Comment:
Now `isRemembered` boolean is not enough because you don't know in which remembered set is. And is possible for an object to be in both.

We should measure if this affect performance and maybe improve the strategy.